### PR TITLE
Fixes #24406 - Modules in a repo

### PR DESCRIPTION
**Pull request by parthaa**
This PR contains changes to show modules in a Repository and content view version detail page. 

managemodules

cvv

### CODE REVIEW

**Comment by johnpmitsch** [comment]
Just re-iterating my comment here since the other PR was closed. If it makes sense to combine the tables into a common template, that would be helpful, especially since we have at least one more page to add this same table to outside this PR. I haven't checked if this makes sense, just an initial thought :)

**Comment by parthaa** [comment]
> Just re-iterating my comment here since the other PR was closed. If it makes sense to combine the tables into a common template, that would be helpful, especially since we have at least one more page to add this same table to outside this PR. I haven't checked if this makes sense, just an initial thought :)

addressed

File: app/models/katello/module_stream.rb [hunk 1]
```diff
@@ -14,7 +14,6 @@ class ModuleStream < ApplicationRecord
     scoped_search on: :context, complete_value: true
     scoped_search on: :arch, complete_value: true
     scoped_search relation: :repositories, on: :name, rename: :repository, complete_value: true
-    scoped_search relation: :repositories, on: :id, rename: :repository_id, complete_value: true
```
Comment (johnpmitsch) [inline app/models/katello/module_stream.rb]: I'm still seeing two links in the UI using this search
```
engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/views/content-view-repositories.html:            <a ng-href="/module_streams?search=repository_id={{repository.id}}">
engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/versions/views/content-view-version-yum.html:            <a ng-href="/module_streams?search=repository_id={{repository.id}}">
```

Comment (parthaa) [reply app/models/katello/module_stream.rb]: nice catch!. Addressed

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-manage-module-streams.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<span page-title ng-model="repository">{{ 'Manage Module Streams for Repository:' | translate }} {{ repository.name }}</span>
```
Comment (johnpmitsch) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-manage-module-streams.html]: where is this page? I can't seem to find a way to get to it

**Comment by johnpmitsch** [comment]
works well and looks good, one small question

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-manage-module-streams.html (hunk 2 above)
Comment (parthaa) [reply engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/views/repository-manage-module-streams.html]: https://github.com/Katello/katello/pull/7698/files#diff-ffb10a7f4a2f3444955cfad88307f3c1R150

Click on the module stream counts

**Comment by omkarkhatavkar** [comment]
Issues: After navigating to Content-View -> Versions -> Yum Repository->Click on Module Steam Count. Url seems like odd => **http://localhost:3000/products//repositories/2/content/content/module_streams?page=1&per_page=20**

1. There are two slashes in url 
2. If you navigate to Repositories link there, No repositories are shown. 
3. Navigate to Content-View -> Versions ->Module Stream, why we are showing all repository here? why only those are specific that version?
4. If it is ok statement 3, seeing weird issue navigate back and forth on all repository with yum repository and seeing duplicate drop down options.   

**Issue 1**
Step 1 : 
image

Step 2: 
image

**Issue 2** 
image

**Comment by omkarkhatavkar** [comment]
One Another Issue 
URL  => http://localhost:3000/products/all/repositories/57/content/content/module_streams?page=1&per_page=20

image

**Comment by johnpmitsch** [comment]
I'm not able to reproduce the double slash or 'repositories' link issue, but I do see a URL discrepancy clicking a module count in a repo (on the product repo list page).

For module counts:
`/products/41/repositories/9/content/content/module_streams?page=1&per_page=20`

For packages:
`/products/41/repositories/9/content/packages?page=1&per_page=20`

Notice the`content/content` vs `content/`

@omkarkhatavkar what did you do to create the 'Couldn't find product all' error message? The links are going to an individual product for me.

**Comment by omkarkhatavkar** [comment]
@johnpmitsch can you please take look at this. Please let me know if db issue i'm having. 

ezgif com-video-to-gif

**Comment by johnpmitsch** [comment]
@omkarkhatavkar I actually get the same error - I didn't realize it was from the content view version add/remove repositories tab. Nice catch! @parthaa can you look?

**Comment by omkarkhatavkar** [comment]
@johnpmitsch @parthaa here is another one GIF 

ezgif com-video-to-gif 2

**Comment by parthaa** [comment]
@johnpmitsch 

> I'm not able to reproduce the double slash or 'repositories' link issue, but I do see a URL discrepancy clicking a module count in a repo (on the product repo list page).

resolved

**Comment by parthaa** [comment]
> URL => http://localhost:3000/products/all/repositories/57/content/content/module_streams?page=1&per_page=20

resolved

**Comment by parthaa** [comment]
@omkarkhatavkar 
> @johnpmitsch can you please take look at this. Please let me know if db issue i'm having.

resolved

**Comment by parthaa** [comment]
> 4\. If it is ok statement 3, seeing weird issue navigate back and forth on all repository with yum repository and seeing duplicate drop down options.

All Repositories - duplication issue should now be addressed

**Comment by parthaa** [comment]
[test katello]

**Comment by omkarkhatavkar** [comment]
Great Work @parthaa :+1: 

**Scenario's Passed:**
1. Module Stream GET, Count works fine on Content Views-> Yum Content page - Passed
2. Module Stream Search works fine - Passed 
3. Module Stram GET, Count works fine on http://localhost/content_views/6/versions/65/yum?page
4. Module Stream GET, Count works fine on the Product page - Passed
5. Module Stream GET, Count work fine in http://localhost:3000/module_streams page -- Passed 

**Scenario's Failed :**
1. Still seeing // after navigating from http://localhost/content_views/6/versions/65/yum?page -> click on module streams 

**Comments:**
The duplicate options got removed on the page, but fix shown the blank filters on initial click, is that what we are intending ?

**Comment by akofink** [comment]
I'm also seeing the double slash (//) @omkarkhatavkar points out.

**Comment by omkarkhatavkar** [comment]
@parthaa Looks great!
double slash issue went away, now shows correct resources.

**Comment by parthaa** [comment]
> The duplicate options got removed on the page, but fix shown the blank filters on initial click, is that what we are intending ?

@omkar @johnpmitsch @akofink  this behavior should be fixed

**Comment by omkarkhatavkar** [comment]
Nice Work @parthaa! looks great. all mentioned issues are closed :+1:

**Pull request merged**